bufferserver/src/main/java/com/datatorrent/bufferserver/internal/LogicalNode.java

```diff
@@ -293,8 +294,9 @@ public boolean addedData()
               }
             }
           }
-        } catch (InterruptedException ie) {
-          throw new RuntimeException(ie);
+        } catch (Exception e) {
```

Comment:
To handle both `InterruptedException` and `RuntimeException`. `addedData` is executed by `autoFlushExecutor` as Runnable and exceptions are not propagated (and they should not, Publisher will not be able to handle them anyway). All exceptions need to be handled by addedData() or by the Runnable itself, otherwise, any exception not handled here will lead to the worker thread termination.